### ISSUE TITLE
Enhancement: Update TB's "Move to Again" and "Copy to Again" lists with move and copy onto a QuickFolders tab

### ISSUE THREAD

**Issue by ptlambert** [RealRaven2000/QuickFolders#291]
I am using QuickFolders v5.11 with Thunderbird 91.11.0 (64-bit) on Windows 11.

When I drag a mail message onto a QuickFolders tab the message moves fine.  However, "Move to <last folder> Again" in the Thunderbird message context menu is not being updated to reflect the folder that the message was moved to.

Similar behavior exist with Ctrl-click-and-drag.  The message is copied OK by QuickFolders into the target tab/folder, but "Copy to <last folder> Again" in the Thunderbird message context menu is not being updated to reflect the folder that the message was copied to.

Would be nice to have Thunderbird's message context menu reflect QuickFolders move and copy activity.  Thanks...

**Comment by RealRaven2000** [comment]
I did some investigation in the meantime (since the OP mentioned the problem in an email) and found that Thunderbird is storing this information in about:config, into these 2 global configuration settings:

```
mail.last_msg_movecopy_target_uri
last_msg_movecopy_was_move
```

This is set when an email is dropped in the folder tree (which explains why it didn't so far show up when we drop mails on a QuickFolders tab or use the quickMove function:
https://searchfox.org/comm-esr102/source/mail/base/content/folderPane.js#1050

**Comment by RealRaven2000** [comment]
Here is a test version that sets both values after using my internal function QuickFolders.Util.moveMessages()

[QuickFolders-mx-5.11.1pre12.zip](https://github.com/RealRaven2000/QuickFolders/files/9236483/QuickFolders-mx-5.11.1pre12.zip)

To install, download the zip file and drag it into Thunderbird Add-ons manager (do not extract files). Would be interested whether it works in Thunderbird 91 as well.

**Comment by ptlambert** [comment]
That build does update TB's "Move to <last folder> Again" list when I 
drag a message onto a QuickFolders tab.  However it also updates the 
/same /list when I Ctrl-click-drag a message onto a QuickFolders tab.  
In that case Thunderbird should present the "Copy to <last folder> 
Again" list, but it's not doing that.

**Comment by RealRaven2000** [comment]
> That build does update TB's "Move to <last folder> Again" list when I drag a message onto a QuickFolders tab.  However it also updates the /same /list when I Ctrl-click-drag a message onto a QuickFolders tab.  In that case Thunderbird should present the "Copy to <last folder> Again" list, but it's not doing that.

thanks for testing! I would expect it to work because I am setting the "isMove" flag too, which I would imagine to be false when you force copying a message. Will test later (We have Add-on developers meeting now)

**Comment by RealRaven2000** [comment]
> > That build does update TB's "Move to  Again" list when I drag a message onto a QuickFolders tab.  However it also updates the /same /list when I Ctrl-click-drag a message onto a QuickFolders tab.  In that case Thunderbird should present the "Copy to  Again" list, but it's not doing that.
> 
> thanks for testing! I would expect it to work because I am setting the "isMove" flag too, which I would imagine to be false when you force copying a message. Will test later (We have Add-on developers meeting now)
> 
> Consequently, I would expect it to show "Copy again" instead.

Testing in Tb102 even though I am using Ctrl while dropping the mail it is actually being moved, so here is another bug underneath...

**Comment by RealRaven2000** [comment]
>That build does update TB's "Move to Again" list when I drag a message onto a QuickFolders tab.  However it also updates the /same /list when I Ctrl-click-drag a message onto a QuickFolders tab.  In that case Thunderbird should present the "Copy to Again" list, but it's not doing that.

I found the problem - I the setting stored in the wrong key, 
it should have been `mail.last_msg_movecopy_was_move`

[QuickFolders-mx-5.11.1pre13.zip](https://github.com/RealRaven2000/QuickFolders/files/9237617/QuickFolders-mx-5.11.1pre13.zip)

To install, download the zip file and drag it into Thunderbird Add-ons manager (do not extract files). Would be interested whether it works in Thunderbird 91 as well.

**Comment by ptlambert** [comment]
QuickFolders-mx-5.11.1pre13.zip seems to be working correctly for me.  
Click-and-drag of a message to a QuickFolders tab causes Thunderbird to 
display "Move to /folder/ Again" for the corresponding folder in the 
message context menu.  Also, Ctrl-click-drag of a message to a 
QuickFolders tab causes Thunderbird to display "Copy to /folder/ Again" 
item in the message context menu.  Thanks for your work on this.

**Comment by RealRaven2000** [comment]
Implemented in v5.12 - Published 17/08/2022
